### ISSUE TITLE
`kubectl logs` with `-l` flag doesn't print all logs with `--tail=-1`

### ISSUE THREAD

**Issue by alexfedoseev** [kubernetes/kubernetes#79967]
**What happened**:
When I run `kubectl logs -l app=my-label --tail=-1` it prints last 10 lines from each pod (30 lines for 3 pods in my case).

**What you expected to happen**:
All lines from all logs should be printed.

**How to reproduce it (as minimally and precisely as possible)**:
I guess by running this command against cluster with multiple pods with the same label.

**Anything else we need to know?**:
When I run `kubectl logs -l app=my-label --tail=1000` it prints 1000 lines.

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.2", GitCommit:"66049e3b21efe110454d67df4fa62b08ea79a19b", GitTreeState:"clean", BuildDate:"2019-05-16T18:55:03Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.1", GitCommit:"b7394102d6ef778017f2ca4046abbaa23b88c290", GitTreeState:"clean", BuildDate:"2019-04-08T17:02:58Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
```

- Cloud provider or hardware configuration: DigitalOcean
- OS (e.g: `cat /etc/os-release`): client is on `macOS 10.14.5`, server nodes are on `Debian GNU/Linux 9 (stretch)`
- Kernel (e.g. `uname -a`): `Darwin alex-mb-pro.local 18.6.0 Darwin Kernel Version 18.6.0: Thu Apr 25 23:16:27 PDT 2019; root:xnu-4903.261.4~2/RELEASE_X86_64 x86_64`
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

**Comment by alenkacz** [comment]
@alexfedoseev `--tail=-1: Lines of recent log file to display. Defaults to -1 with no selector, showing all log lines otherwise 10, if a selector is provided.`

that sounds to me like it's actually using the default even if you explicitly specified `-1`

**Comment by alenkacz** [comment]
I can take a look at this one if you're not planning on submitting PR..

**Comment by alexfedoseev** [comment]
Thanks! I also try to use `--since-time` flag but it seems it also ignored (i.e. I receive default 10 lines per pod).

**Comment by alenkacz** [comment]
Hi, I looked into it and it seems to be already fixed on master #74900
